src/graphql/fanout.ts

```diff
@@ -0,0 +1,307 @@
+import {
+  nonNull,
+  mutationField,
+  arg,
+  objectType,
+  extendInputType,
+  list,
+} from 'nexus';
+import { YogaInitialContext } from 'graphql-yoga';
+import { decryptEncodedPayload } from '../lib/cryptography/utils.js';
+import { FanoutClient, MembershipModel } from '@glasseaters/hydra-sdk';
+import { Connection, clusterApiUrl, Keypair, PublicKey } from '@solana/web3.js';
+import { NATIVE_MINT } from '@solana/spl-token';
+import { Wallet } from '@project-serum/anchor';
+
+export const SplFanout = objectType({
+  name: 'SplFanout',
+  description: 'The spl fanout result',
+  definition (t) {
+    t.nonNull.string('splTokenAddress', {
+      description: 'SPL Token Address',
+    });
+    t.nonNull.string('splTokenWallet', {
+      description: 'SPL Token Wallet',
+    });
+  }
+})
+
+export const CreateFanoutResult = objectType({
+  name: 'CreateFanoutResult',
+  description: 'The result for minting a NFT',
+  definition (t) {
+    t.nonNull.string('message', {
+      description: 'Operation message',
+    });
+    t.string('fanoutPublicKey', {
+      description: 'Fanout public key',
+    });
+    t.field('solanaWalletAddress', {
+      type: 'String',
+      description: 'Solana address of the fanout',
+    });
+    t.field('splFanout', {
+      type: 'SplFanout',
+      description: "Spl Fanout Details"
+    })
+    t.field('splFanoutAddress', {
+      type: 'String',
+      description: 'Spl address of the fanout',
+    });
+    t.field('splWalletAddress', {
+      type: 'String',
+      description: 'Spl address of the fanout',
+    });
+  },
+});
+
+export const DisperseFanoutResult = objectType({
+  name: 'DisperseFanoutResult',
+  description: 'The result for minting a NFT',
+  definition (t) {
+    t.nonNull.string('message', {
+      description: 'Operation message',
+    });
+  },
+});
+
+export const FanoutMember = extendInputType({
+  type: 'FanoutMember',
+  definition (t) {
+    t.nonNull.field('publicKey', {
+      type: 'String',
+      description: 'Public key of member address',
+    });
+    t.nonNull.field('shares', {
+      type: 'Int',
+      description: 'Share member should receive',
+    });
+  },
+});
+
+export const CreateFanout = mutationField('createFanout', {
+  type: 'CreateFanoutResult',
+  args: {
+    encryptedMessage: nonNull(
+      arg({
+        type: 'EncryptedMessage',
+      }),
+    ),
+    name: nonNull(
+      arg({
+        type: 'String',
+      }),
+    ),
+    members: nonNull(
+      arg({
+        type: list('FanoutMember'),
+      }),
+    ),
+    splTokenAddresses: arg({
```

Comment:
It took me a bit to understand how this worked to maybe we could add a description like:
"Token mint addresses for which the fanout will create token accounts to distribute from, leave empty for a native SOL only fanout"

Reply:
i was thinking about this as well, you could specify the SOL token address in the list and a SOL disbursement will also take place - its not the clearest i agree